### PR TITLE
Run all unit tests other than vendor code

### LINKED ISSUE #1159: `make test` not running all unit tests
For whatever reason the make test target was being very specific about the test directories used - this has resulted in that target not running all unit tests in the code.
#1158 is one example of the result.

### PR DESCRIPTION
**Pull request by hickeng**
We're currently whitelisting unit test directories which is resulting in some tests getting skipped. This whitelists the top level repo directories with project code in, exempting the vendor directory.

Fixes #1159 #1299

### CODE REVIEW

**Comment by caglar10ur** [comment]
It looks like this PR also have some other changes in it and some commits does not match with their description, see https://github.com/vmware/vic/pull/1294/commits/70dd4cace4dff1a79f96e11559831ac8d990922b it says "Run all unit tests excluding vendor" but adds a new file and does nothing else. 😕

**Comment by caglar10ur** [comment]
Oh so you are really moving code around? It's not just a makefile change.

**Comment by hickeng** [comment]
@caglar10ur yup - needed to actually move code around to make the tests work properly. We had cyclic package inclusion errors in one of the unit tests.

**Comment by caglar10ur** [comment]
LGTM :sheep: :it:

**Comment by hickeng** [comment]
poke

**Comment by hickeng** [comment]
@dougm @fdawg4l can I get a review so the next time this passes it can merge.

**Comment by dougm** [comment]
lgtm

File: Makefile [hunk 1]
```diff
@@ -190,14 +190,8 @@ vendor: $(GVT)
...
-TEST_DIRS=github.com/vmware/vic/cmd/tether
-TEST_DIRS+=github.com/vmware/vic/cmd/imagec
-TEST_DIRS+=github.com/vmware/vic/cmd/vicadmin
-TEST_DIRS+=github.com/vmware/vic/cmd/rpctool
-TEST_DIRS+=github.com/vmware/vic/cmd/vic-machine
-TEST_DIRS+=github.com/vmware/vic/lib/apiservers/portlayer
-TEST_DIRS+=github.com/vmware/vic/lib/install
-TEST_DIRS+=github.com/vmware/vic/lib/portlayer
+TEST_DIRS=github.com/vmware/vic/cmd
+TEST_DIRS+=github.com/vmware/vic/lib/
```
Comment (dougm) [inline Makefile]: cosmetic, but only this path has a trailing '/'

**Comment by corrieb** [comment]
@hickeng is there a reason we're waiting to merge this?

**Comment by jzt** [comment]
@hickeng I have #1613 that appears to be a subset of this task. I've checked out your branch, rebased it against master, added my changes and have no begun integration testing. Do you mind if I take this issue and open a new PR with it?

**Comment by jzt** [comment]
@hickeng ok well it seems like you're back on this so I'll ditch my changes and move onto something else.

**Comment by hickeng** [comment]
poke

**Comment by hickeng** [comment]
poke

**Pull request merged**